crates/wasmparser/src/validator/types.rs

```diff
@@ -1003,46 +999,35 @@ impl<'a> TypesRef<'a> {
         }
     }
 
-    fn types(&self, core: bool) -> Option<&'a [TypeId]> {
-        Some(match &self.kind {
-            TypesRefKind::Module(module) => {
-                if core {
-                    &module.types
-                } else {
-                    return None;
-                }
-            }
-            TypesRefKind::Component(component) => {
-                if core {
-                    &component.core_types
-                } else {
-                    &component.types
-                }
-            }
-        })
-    }
-
     /// Gets a type based on its type id.
     ///
     /// Returns `None` if the type id is unknown.
-    pub fn type_from_id(&self, id: TypeId) -> Option<&'a Type> {
+    pub fn get(&self, id: TypeId) -> Option<&'a Type> {
         self.list.get(id.index)
     }
 
     /// Gets a type id from a type index.
```

Comment:
This comment should now at least say "core" type, and mention the index is a "core type index", to distinguish it from the comment for `component_type_at`.